nntrainer/tensor/tensor_wrap_specs.h

```diff
@@ -70,6 +74,40 @@ typedef std::tuple<TensorDim, Tensor::Initializer, bool, const std::string,
                    TensorLifespan>
   VarGradSpec;
 
+/**
+ * @brief Tensor Specification which describes how this tensor should be
+ * allocated and managed
+ *
+ */
+struct TensorSpecV2 {
+  enum class RequestType {
+    PLACEHOLDER,
+    UNIQUE,
```

Comment:
maybe UNIQUE -> NEW? All tensors are themselves unique anyway but sharing internal memory structure.

Reply:
This is to represent the ownership semantics of `UNIQUE_`ptrvs `SHARED_`ptr, I don't mind NEW although please review this idea once again :)